File: epi_judge_java/epi/SortedListsMerge.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class SortedListsMerge {

    @EpiTest(testDataFile = "sorted_lists_merge.tsv")
    public static ListNode<Integer> mergeTwoSortedLists(ListNode<Integer> list1, ListNode<Integer> list2) {
        ListNode<Integer> dummy = new ListNode<>(-1, null);
        ListNode<Integer> runner = dummy;
        while (list1 != null && list2 != null) {
            if (list1.data < list2.data) {
                runner.next = list1;
                list1 = list1.next;
            } else {
                runner.next = list2;
                list2 = list2.next;
            }
            runner = runner.next;
        }
        if (list1 == null) runner.next = list2;
        if (list2 == null) runner.next = list1;
        return dummy.next;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "SortedListsMerge.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/sorted_lists_merge.cc
#include "list_node.h"
#include "test_framework/generic_test.h"

using namespace std;

shared_ptr<ListNode<int>> MergeTwoSortedLists(shared_ptr<ListNode<int>> list1,
                                              shared_ptr<ListNode<int>> list2) {

    auto dummy = make_shared<ListNode<int>>(-1);
    auto runner = dummy;
    while (list1 != nullptr || list2 != nullptr) {
        if (list2 == nullptr || (list1 != nullptr && list1->data < list2->data)) {
            runner->next = make_shared<ListNode<int>>(list1->data);
            list1 = list1->next;
        } else {
            runner->next = make_shared<ListNode<int>>(list2->data);
            list2 = list2->next;
        }
        runner = runner->next;
    }

    return dummy->next;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"L1", "L2"};
    return GenericTestMain(args, "sorted_lists_merge.cc",
                           "sorted_lists_merge.tsv", &MergeTwoSortedLists,
                           DefaultComparator{}, param_names);
}
